tests/store/tracking/test_sqlalchemy_store.py

```diff
@@ -537,7 +534,9 @@ def test_run_needs_uuid(self):
         # Depending on the implementation, a NULL identity key may result in different
         # exceptions, including IntegrityError (sqlite) and FlushError (MysQL).
         # Therefore, we check for the more generic 'SQLAlchemyError'
-        with self.assertRaises(MlflowException) as exception_context:
+        with self.assertRaisesRegex(
+            MlflowException, r"NOT NULL constraint failed"
+        ) as exception_context:
```

Comment:
We run `test_sqlalchemy_store.py` against MySQL, SQLServer, PostgreSQL, and SQLite. We need to use a different message matcher for each database:

https://github.com/mlflow/mlflow/runs/6444433813?check_suite_focus=true#step:4:7172

Reply:
Thanks for sharing the link. It's really helpful!